Testing/SystemTests/tests/analysis/PearlPowderDiffractionScriptTest.py

```diff
@@ -0,0 +1,190 @@
+            self.assertEquals(1, van_file.getNumberHistograms())
+            self.assertEquals(4310, van_file.blocksize())
+
+        # check random values, places variables doesnt seem to work when
+        van1 = mtd["van1"]
+        self.assertAlmostEqual(1690.17031579, van1.readX(0)[199], places=DIFF_PLACES)
+        van4 = mtd["van4"]
+        self.assertAlmostEqual(6.5605707095e-06, van4.readY(0)[4000], places=DIFF_PLACES)
+        van7 = mtd["van7"]
+        self.assertAlmostEquals(1.03406081396e-08, van7.readE(0)[50], places=DIFF_PLACES)
+        van10 = mtd["van10"]
+        self.assertAlmostEquals(11667.996525642744, van10.readX(0)[3420], places=DIFF_PLACES)
+        van14 = mtd["van14"]
+        self.assertAlmostEqual(3.39159292684e-05, van14.readY(0)[187], places=DIFF_PLACES)
+
+    def test_rdata_workspace(self):
+        for i in range(1, 15, 3):
+            rfile = mtd["rdata" + str(i)]
+            self.assertTrue(isinstance(rfile, MatrixWorkspace))
+            self.assertEquals(1, rfile.getNumberHistograms())
+            self.assertEquals(4310, rfile.blocksize())
+            self.assertIn("glucose+Pb 6 tns in ZTA anvils", rfile.getTitle())
+
+            van_file = mtd["van" + str(i)]
+            self.assertAlmostEquals(van_file.readX(0)[i], rfile.readX(0)[i], places=DIFF_PLACES)
+
+        focus_file = mtd["focus"]
+        self.assertTrue(isinstance(focus_file, MatrixWorkspace))
+        self.assertEquals(14, focus_file.getNumberHistograms())
+        self.assertEquals(4310, focus_file.blocksize())
+        self.assertIn("glucose+Pb 6 tns in ZTA anvils", rfile.getTitle())
+
+    def test_mod_files(self):
+        mod_group_table = mtd["PRL92476_92479"]
+        self.assertTrue("PRL92476_92479_mods1-9_1" in mod_group_table[0].getName())
+
+        rand_range = random.randint(1, 4)
+        for i in range(rand_range, 10, 2):
+            self.assertTrue("PRL92476_92479_mod" + str(i) in mod_group_table[i].getName())
+            self.assertEquals(4310, mod_group_table[i].blocksize())
+            ind_mod_file = mtd["PRL92476_92479_mod" + str(i)]
+            rand_test = random.randint(1, 4300)
+            self.assertAlmostEqual(mod_group_table[i].readY(0)[rand_test],
+                                   ind_mod_file.readY(0)[rand_test], places=DIFF_PLACES)
+
+        no_atten = mtd["PRL92476_92479_noatten"]
+        self.assertAlmostEqual(1.06842521, no_atten.readY(0)[17], places=DIFF_PLACES)
+        self.assertAlmostEqual(1.52553392, no_atten.readX(0)[2663], places=DIFF_PLACES)
+        self.assertAlmostEqual(0.01567764, no_atten.readE(0)[577], places=DIFF_PLACES)
+        self.assertAlmostEqual(3.61208767, no_atten.readX(0)[4100], places=DIFF_PLACES)
+        self.assertAlmostEqual(0.94553930, no_atten.readY(0)[356], places=DIFF_PLACES)
+
+    def test_saved_data_files(self):
+        files_data = []
+
+        wsname = "GSSFile"
+        gssfile = (DIRS[0] + "PEARL/DataOut/PRL92476_92479-0.gss")
```

Comment:
For this kind of `+` that really means append path components or subdirectories it's usually safer and cross-platform to use `os.path.join`